test/integration/cni/pod_traffic_across_az_test.go

```diff
@@ -0,0 +1,305 @@
+package cni
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/integration/common"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	coreV1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/apps/v1"
+)
+
+// Tests pod networking across AZs. It similar to pod connectivity test, but launches a daemonset, so that
+// there is a pod on each node across AZs. It then tests connectivity between pods on different nodes across AZs.
+var _ = Describe("[STATIC_CANARY] test pod networking", FlakeAttempts(3), func() {
+
+	var (
+		err        error
+		serverPort int
+		protocol   string
+
+		// The command to run on server pods, to allow incoming
+		// connections for different traffic type
+		serverListenCmd []string
+		// Arguments to the server listen command
+		serverListenCmdArgs []string
+
+		// The function that generates command which will be sent from
+		// tester pod to receiver pod
+		testConnectionCommandFunc func(serverPod coreV1.Pod, port int) []string
+
+		// The functions re-inforces that the positive test is working as
+		// expected by creating a negative test command that should fail
+		testFailedConnectionCommandFunc func(serverPod coreV1.Pod, port int) []string
+
+		// Expected stdout from the exec command on testing connection
+		// from tester to server
+		testerExpectedStdOut string
+		// Expected stderr from the exec command on testing connection
+		// from tester to server
+		testerExpectedStdErr string
+
+		// Daemonset to run on node
+		testDaemonSet *v1.DaemonSet
+
+		// Map of Pods placed on the node
+		interfaceToPodList common.InterfaceTypeToPodList
+
+		// Map of AZ name, string to pod IP, string
+		azName    string
+		azToPodIP map[string]string
+		azToPod   map[string]coreV1.Pod
+	)
+
+	JustBeforeEach(func() {
+		By("authorizing security group ingress on instance security group")
+		err = f.CloudServices.EC2().
```

Comment:
Why do we need to modify security groups here?

Reply:
TCP connectivity test, ensure that pod is open for egress and ingress.

Reply:
Same Q. Why do we need this? It should work by default.

Reply:
Do we run these tests on v6 clusters?